whois-query/src/main/java/net/ripe/db/whois/query/acl/HazelcastPersonalObjectAccounting.java

```diff
@@ -71,21 +49,28 @@ public int getQueriedPersonalObjects(final InetAddress remoteAddress) {
 
     @Override
     public int accountPersonalObject(final InetAddress remoteAddress, final int amount) {
+        boolean isLocked = false;
+
         try {
-            Integer count = counterMap.tryLockAndGet(remoteAddress, 3, TimeUnit.SECONDS);
+            if (isLocked = counterMap.tryLock(remoteAddress,3, TimeUnit.SECONDS)) {
+                Integer count = counterMap.get(remoteAddress);
+                count = (count == null) ? amount : (count + amount);
+                counterMap.put(remoteAddress, count);
 
-            if (count == null) {
-                count = amount;
-            } else {
-                count += amount;
+                return count;
             }
 
-            counterMap.putAndUnlock(remoteAddress, count);
-            return count;
-        } catch (TimeoutException | IllegalStateException e) {
+            //if cannot get a lock in specified time, return the current state not zero
+            return counterMap.get(remoteAddress);
+
+        } catch (Exception e) {
             LOGGER.info("Unable to account personal object, allowed by default. Threw {}: {}", e.getClass().getName(), e.getMessage());
+        } finally {
+            //unlock only if it is locked by this instance
+            if(isLocked) {
+                counterMap.unlock(remoteAddress);
```

Comment:
Do you need a try/catch here also?

Reply:
To be safe I will add.

Reply:
Actually It can throw Interuptted exception